server/services/starUpgrade.js

```diff
@@ -42,7 +42,8 @@ module.exports = class StarUpgradeService extends EventEmitter {
 
         const expenseConfig = game.constants.star.infrastructureExpenseMultipliers[game.settings.specialGalaxy.warpgateCost];
         const terraformedResources = this.starService.calculateTerraformedResources(star.naturalResources, effectiveTechs.terraforming);
-        const cost = this.calculateWarpGateCost(game, expenseConfig, terraformedResources);
+        let averageTerraformedResources = ( terraformedResources.economy + terraformedResources.industry + terraformedResources.science ) / 3;
```

Comment:
Could be moved into a `starResourceService`? `getAverageTerraformedResources`?

Reply:
It feels this would still fit in the starUpgrade service... As this is the place where it would be used

Reply:
I didnt move it into starResourceService, but I made it a seperate function and cleaned up with that